File: module-server/src/main/java/de/twomartens/wahlrecht/model/VotingResult.java
package de.twomartens.wahlrecht.model;

import java.util.Map;
import lombok.Builder;
import lombok.Getter;

@Getter
public class VotingResult {

  private final int votesOnNomination;
  private final int votesThroughHealing;
  private final Map<Integer, Integer> votesPerPosition;
  private Integer totalVotes = null;

  @Builder
  public VotingResult(int votesOnNomination, int votesThroughHealing,
      Map<Integer, Integer> votesPerPosition) {
    this.votesOnNomination = votesOnNomination;
    this.votesThroughHealing = votesThroughHealing;
    this.votesPerPosition = votesPerPosition;
  }

  public int getTotalVotes() {
    if (totalVotes == null) {
      totalVotes = votesOnNomination + votesThroughHealing + votesPerPosition.values().stream()
          .reduce(0, Integer::sum);
    }
    return totalVotes;
  }
}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/model/db/VotingResult.kt
package de.twomartens.wahlrecht.model.db;

import java.util.Map;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@Builder
@EqualsAndHashCode
@ToString
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor(force = true, access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class VotingResult {
  String electionName;
  String partyAbbreviation;
  String nominationName;
  int votesOnNomination;
  int votesThroughHealing;
  Map<Integer, Integer> votesPerPosition;
}
